pkg/csi/service/wcp/controller_helper.go

```diff
@@ -950,3 +962,74 @@ func GetZonesFromAccessibilityRequirements(ctx context.Context,
 	}
 	return zones, nil
 }
+
+// getSnapshotLimitForNamespace reads the snapshot limit from ConfigMap in the namespace.
+// Returns the effective limit after applying defaults and absolute max clamping.
+func getSnapshotLimitForNamespace(ctx context.Context, namespace string) (int, error) {
+	log := logger.GetLogger(ctx)
+
+	// Get Kubernetes config
+	cfg, err := getK8sConfig()
+	if err != nil {
+		// If k8s config is not available (e.g., in test environments), use default
+		log.Infof("getSnapshotLimitForNamespace: failed to get Kubernetes config, using default: %d. Error: %v",
+			common.DefaultMaxSnapshotsPerVolume, err)
+		return common.DefaultMaxSnapshotsPerVolume, nil
+	}
+
+	// Create Kubernetes clientset
+	k8sClient, err := newK8sClientFromConfig(cfg)
+	if err != nil {
+		// If k8s client creation fails, use default
+		log.Infof("getSnapshotLimitForNamespace: failed to create Kubernetes client, using default: %d. Error: %v",
+			common.DefaultMaxSnapshotsPerVolume, err)
+		return common.DefaultMaxSnapshotsPerVolume, nil
+	}
+
+	// Get ConfigMap from the namespace
+	cm, err := k8sClient.CoreV1().ConfigMaps(namespace).Get(ctx, common.ConfigMapCSILimits, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// ConfigMap not found, use default
+			log.Infof("getSnapshotLimitForNamespace: ConfigMap %s not found in namespace %s, using default: %d",
+				common.ConfigMapCSILimits, namespace, common.DefaultMaxSnapshotsPerVolume)
+			return common.DefaultMaxSnapshotsPerVolume, nil
+		}
+		// Other error occurred
+		log.Errorf("getSnapshotLimitForNamespace: failed to get ConfigMap %s in namespace %s, err: %v",
+			common.ConfigMapCSILimits, namespace, err)
+		return 0, err
+	}
+
+	// Check if the key exists in ConfigMap data
+	limitStr, exists := cm.Data[common.ConfigMapKeyMaxSnapshotsPerVolume]
+	if !exists {
+		// Key not found in ConfigMap, fail the request
+		errMsg := fmt.Sprintf("ConfigMap %s exists in namespace %s but missing required key '%s'",
+			common.ConfigMapCSILimits, namespace, common.ConfigMapKeyMaxSnapshotsPerVolume)
+		log.Errorf("getSnapshotLimitForNamespace: %s", errMsg)
```

Comment:
This seems very strict. Can we fall back to use default values if key is missing?